Objects/unicodeobject.c

```diff
@@ -118,14 +118,14 @@ NOTE: In the interpreter's initialization phase, some globals are currently
     (assert(_PyUnicode_CHECK(op)),                      \
      PyUnicode_IS_COMPACT_ASCII(op) ?                   \
          ((char*)(_PyASCIIObject_CAST(op) + 1)) :       \
-         _PyUnicode_UTF8(op))
+        FT_ATOMIC_LOAD_PTR(_PyUnicode_UTF8(op)))
```

Comment:
I don't think we should put the atomics operations in the macros -- it's too easy to hide data races that way.

`PyUnicode_UTF8()` functions as both a guard and as an accessor. We should split those up.